draft-irtf-cfrg-hybrid-kems.md

```diff
@@ -303,236 +425,231 @@ MUST provide the bit-security required to source input randomness for PQ/T
+constructions share a common overall structure, differing mainly in how they
+compute the final shared secret and the security requirements of their components.
 
-## KEM from Diffie-Hellman {#group}
+## HashEverything
 
-This section describes a simple KEM built from a Diffie-Hellman group.  **This
-KEM is not a secure KEM in the sense of the IND-CCA standard usually applied (it
-meets the lower IND-CPA standard {{ABR01}}), and thus should not be used on its
-own.**  However, using the constructions in this document, it can be used as a
-component of an IND-CCA hybrid KEM, as discussed in {{security-considerations}}.
+The HashEverything hybrid KEM (`KEM_H` below) depends on the following
+constituent components:
 
-The traditional Diffie-Hellman construction depends on an abelian group G with a
-chosen group generator or "base point" `B`, in which the discrete-log problem is
-hard.  Here we write the group operation in G additively, so that "scalar
-multiplication" of a non-negative integer `k` times a group element `P`
-represents repeated application of the group operation to `P`:
+* `KEM_T`: A traditional KEM
+* `KEM_PQ`: A post-quantum KEM
+* `Expand`: A KDF producing byte strings of length `KEM_T.Nseed + KEM_PQ.Nseed`
+  (`Expand.Nout == KEM_T.Nseed + KEM_PQ.Nseed`)
+* `Combine`: A KDF producing byte strings of length `KEM_H.Nss` (`Combine.Nout
+  == KEM_H.Nss`)
+* `Label` - A byte string used to label the specific combination of the above
+  constituents being used.
+
+We presume the KEMs and KDFs meet the interfaces described in
+{{cryptographic-deps}}.
+
+The constants associated with the hybrid KEM are mostly derived from the
+concatenation of keys and ciphertexts:
 
 ~~~
-0 * P = O (the identity element for the group)
-1 * P = P
-2 * P = P + P
-...
-k * P = (k - 1) * P + P
+Nek = KEM_T.Nek + KEM_PQ.Nek
+Ndk = KEM_T.Ndk + KEM_PQ.Ndk
+Nct = KEM_T.Nct + KEM_PQ.Nct
 ~~~
 
-We call a non-negative integer in the range `[0, n)` a "scalar", where `n` is
-the order of the group.
-
-In addition to the group operation, we require that a Diffie-Hellman group
-define the following algorithms:
+The `Nseed` and `Nss` constants should reflect the overall security level of the
+combined KEM, with the following recommended values:
 
-- `RandomScalar(seed) -> k`: Produce a uniform pseudo-random scalar from the
-  byte string `seed`.
-- `ScalarToBytes(k) -> dk`: Encode a scalar into a fixed-length byte string.
-- `BytesToScalar(dk) -> k`: Decode a scalar from a fixed-length byte string.
-- `ElementToBytes(P) -> dk`: Encode an element of the group into a fixed-length
-  byte string.
-- `BytesToElement(dk) -> P`: Decode an element of the group from a fixed-length
-  byte string.
-- `ElementToSharedSecret(P) -> ss`: Extract a shared secret from an element of
-  the group (e.g., by taking the X coordinate of an ellpitic curve point).
+~~~
+Nseed = max(KEM_T.Nseed, KEM_PQ.Nseed)
+Nss = min(KEM_T.Nss, KEM_PQ.Nss)
+~~~
```

Comment:
@bwesterb @br-hale @pag-crypto 

I feel like there's a little bit more to this than mix/max'ing but I'm not sure